src/core/macro.scala

```diff
@@ -259,12 +281,223 @@ object Macro:
         .groupBy { case (name, _) => name }
         .toList
         .map { case (name, l) => name -> l.flatMap(_._2) }
-        .map { (name, anns) => Expr.ofTuple(Expr(name), Expr.ofList(anns)) }
+        .map { case (name, anns) => Expr.ofTuple(Expr(name), Expr.ofList(anns)) }
 
     private def isObjectOrScala(bc: Symbol) =
       bc.name.contains("java.lang.Object") || bc.fullName.startsWith("scala.")
 
     private def filterAnnotation(a: Term): Boolean =
       a.tpe.typeSymbol.maybeOwner.isNoSymbol ||
         a.tpe.typeSymbol.owner.fullName != "scala.annotation.internal"
-  }
+
+object MirrorlessMacro:
```

Comment:
aren't the other macros mirrorless as well? maybe let's just group by functionality, `ValueClassDerivation` seems good :)

plus, can we make it private? I know the others are not, they should probably be, but let's not increase the API surface area if we don't have to

Reply:
I have nested a somehow `ValueClassDerivation` object inside a more general `Macro` object. Macro implementations for VC have been made private. The already existing one cannot be marked private without compatibility issues though.